data/ha/hawk_test_ssh.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/python3
+"""Define SSH related functions to test the HAWK GUI"""
+
+from distutils.version import LooseVersion as Version
+import paramiko, hawk_test_results, warnings
+# Ignore CryptographyDeprecationWarning shown when using paramiko
+try:
+    from cryptography.utils import CryptographyDeprecationWarning
+    warnings.simplefilter('ignore', CryptographyDeprecationWarning)
+except ImportError:
+    pass
+
+class hawkTestSSHError(Exception):
+    """Base class for exceptions in this module."""
+    def __init__(self, value):
+        self.value = value
+
+    def __str__(self):
+        return repr(self.value)
+
+class hawkTestSSH:
+    def __init__(self, hostname, secret):
+        self.ssh = paramiko.SSHClient()
+        self.ssh.load_system_host_keys()
+        self.ssh.set_missing_host_key_policy(paramiko.WarningPolicy)
+        self.ssh.connect(hostname=hostname.lower(), username="root", password=secret)
+
+    def is_ssh(self):
+        clase = str(type(self.ssh)).split(' ')[1][1:26]
+        if clase != 'paramiko.client.SSHClient':
+            raise hawkTestSSHError('SSH object must be of type %s. Got: [%s]' %
+                                   ('paramiko.client.SSHClient', type(s)))
+        return True
+
+    def check_cluster_conf_ssh(self, command, mustmatch):
+        command = str(command)
+        if self.is_ssh():
+            resp = self.ssh.exec_command(command)
+            out = resp[1].read().decode().rstrip('\n')
+            err = resp[2].read().decode().rstrip('\n')
+            print("INFO: ssh command [%s] got output [%s] and error [%s]" % (command, out, err))
+            if err:
+                print("ERROR: got an error over SSH: [%s]" % err)
+                return False
+            if isinstance(mustmatch, str):
+                if mustmatch:
+                    if out.find(mustmatch) >= 0:
+                        return True
+                    return False
+                return out == mustmatch
+            elif isinstance(mustmatch, list):
+                for exp in mustmatch:
+                    if out.find(str(exp)) < 0:
+                        return False
+                return True
+            else:
+                raise hawkTestSSHError("check_cluster_conf_ssh: mustmatch must be str or list")
+        return False
+
+    def set_test_status(self, results, test, status):
+        if isinstance(results, hawk_test_results.resultSet):
+            results.set_test_status(test, status)
+
+    def verify_stonith_in_maintenance(self, results):
+        if self.check_cluster_conf_ssh("crm status | grep stonith-sbd", "unmanaged"):
+            print("INFO: stonith-sbd is unmanaged")
+            self.set_test_status(results, 'verify_stonith_in_maintenance', 'passed')
+            return 0
+        print("ERROR: stonith-sbd is not unmanaged but should be")
+        self.set_test_status(results, 'verify_stonith_in_maintenance', 'failed')
+        return 30    # return non zero value on error
```

Comment:
maybe you can create some Error enum which would contains this magic digits bind to some human readable string ? 
There is two options actually - one is you actually don't care about certain number in this case just return always 1 to not confuse people. other is you do care and in this case you will get lost soon in this numbers so you need to document it in some way and Errors enum would be one option or just some text fail containing all numbers with explanations

Reply:
Agree. Originally the idea was doing something with the return values in the perl module that calls this, but now you are right that it doesn't have rhyme or reason. Will change this to return 1.